src/sentry/api/bases/sentryapps.py

```diff
@@ -99,6 +103,45 @@ def dispatch(self, request, *args, **kwargs):
 class SentryAppsBaseEndpoint(IntegrationPlatformEndpoint):
     permission_classes = (SentryAppsPermission,)
 
+    def _get_organization_slug(self, request):
+        organization_slug = request.json_body.get("organization")
+        if not organization_slug or not isinstance(organization_slug, string_types):
+            error_message = """
+                Please provide a valid value for the 'organization' field.
+            """
+            raise ValidationError({"organization": to_single_line_str(error_message)})
+        return organization_slug
+
+    def _get_organization_for_superuser(self, organization_slug):
+        try:
+            return Organization.objects.get(slug=organization_slug)
+        except Organization.DoesNotExist:
+            error_message = """
+                Organization '{}' does not exist.
+            """.format(
+                organization_slug
+            )
+            raise ValidationError({"organization": to_single_line_str(error_message)})
+
+    def _get_organization_for_user(self, user, organization_slug):
+        try:
+            return user.get_orgs().get(slug=organization_slug)
+        except Organization.DoesNotExist:
+            error_message = """
+                User does not belong to the '{}' organization.
+            """.format(
+                organization_slug
+            )
+            raise PermissionDenied(to_single_line_str(error_message))
+
+    def _get_organization(self, request):
```

Comment:
@jverce This change is definitely an improvement but it won't handle the case of using an integration token correctly (internal integration token). That case happens when `user.is_sentry_app==True`. I'm OK not handling that case but we should probably do a validation check here saying you can't use an integration token. Here's sort of how that check has to work for an integration token:

https://github.com/getsentry/sentry/blob/861531308c5ef4e26d8dfc0e8690b951ea17d17f/src/sentry/auth/access.py#L316-L322

Reply:
Will take a look!

Reply:
Turns out this is already taken care of by the auth middleware in all cases.
I added a few tests for this so that you can validate that I'm addressing your concern or that I'm totally off track :joy:

Reply:
@jverce I think it would be better to handle this case by throwing an error like `Cannot create an integration using an integration token`. The reason being that if an integration had the `org:admin` scope, it really _should_ be able to generate another integration based on current abstractions of checking org membership in the `access.py` file I linked to you. Otherwise it might be confusing to anyone using an integration token why they are getting a permission denied error.